src/main/java/com/amazon/dlic/util/SettingsBasedSSLConfiguratorV4.java

```diff
@@ -0,0 +1,561 @@
+        }
+
+        public String getEffectiveKeyAlias() {
+            return effectiveKeyAlias;
+        }
+
+        public List<String> getEffectiveTruststoreAliases() {
+            return effectiveTruststoreAliases;
+        }
+
+        public String[] getEffectiveTruststoreAliasesArray() {
+            if (this.effectiveTruststoreAliases == null) {
+                return null;
+            } else {
+                return this.effectiveTruststoreAliases.toArray(new String[this.effectiveTruststoreAliases.size()]);
+            }
+        }
+
+        public String[] getEffectiveKeyAliasesArray() {
+            if (this.effectiveKeyAlias == null) {
+                return null;
+            } else {
+                return new String[] { this.effectiveKeyAlias };
+            }
+        }
+
+        @Override
+        public String toString() {
+            return "SSLConfig [sslContext=" + sslContext + ", supportedProtocols=" + Arrays.toString(supportedProtocols)
+                    + ", supportedCipherSuites=" + Arrays.toString(supportedCipherSuites) + ", hostnameVerifier="
+                    + hostnameVerifier + ", startTlsEnabled=" + startTlsEnabled + ", hostnameVerificationEnabled="
+                    + hostnameVerificationEnabled + ", trustAll=" + trustAll + ", effectiveTruststore="
+                    + effectiveTruststore + ", effectiveTruststoreAliases=" + effectiveTruststoreAliases
+                    + ", effectiveKeystore=" + effectiveKeystore + ", effectiveKeyAlias=" + effectiveKeyAlias + "]";
+        }
+
+        public boolean isTrustAllEnabled() {
+            return trustAll;
+        }
+    }
+
+    public static class SSLConfigException extends Exception {
+
+        private static final long serialVersionUID = 5827273100470174111L;
+
+        public SSLConfigException() {
+            super();
+        }
+
+        public SSLConfigException(String message, Throwable cause, boolean enableSuppression,
+                                  boolean writableStackTrace) {
+            super(message, cause, enableSuppression, writableStackTrace);
+        }
+
+        public SSLConfigException(String message, Throwable cause) {
+            super(message, cause);
+        }
+
+        public SSLConfigException(String message) {
+            super(message);
+        }
+
+        public SSLConfigException(Throwable cause) {
+            super(cause);
+        }
+
+    }
+
+    private static class OverlyTrustfulSSLContextBuilder extends SSLContextBuilder {
+        @Override
+        protected void initSSLContext(SSLContext sslContext, Collection<KeyManager> keyManagers,
+                                      Collection<TrustManager> trustManagers, SecureRandom secureRandom) throws KeyManagementException {
+            sslContext.init(!keyManagers.isEmpty() ? keyManagers.toArray(new KeyManager[keyManagers.size()]) : null,
+                    new TrustManager[] { new OverlyTrustfulTrustManager() }, secureRandom);
```

Comment:
This can be dismissed, this is by design.  For testing scenarios the security plugin provides SSL configurations that are more open to make cert creation easier.